### DISCUSSION TITLE
HTML Styles

### DISCUSSION THREAD

**Discussion by systemz64** [mkende/pmarkdown#2] [Ideas]
just implemented this module and it works really well for conversions. Would be really nice to update it to match the cleaner looking code outputs you get with things like ChatGPT. Or the ability to define styles for elements. Otherwise - very nice

**Comment by mkende** [comment]
Do you have an example of the formatting that you are looking for? It's definitely possible to have an option to pass the generated HTML through some existing formatter, but this could also be done externally.

Do you mind, telling me where you are using this module?

**Comment by systemz64** [comment]
Hi Mathias

We have built a chatbot into our application that uses the OpenAI API. We struggled to get the LLMs to respond with decent HTML just using prompts – so we had to accept that they are best suited to Markdown. Hence the need for a convertor.

So here is an example output using your tool:

***@***.***

Certainly works!

Here is the latest on ChatGPT

***@***.***


I suspect they have a custom convertor that understands specific languages – but even “dark” mode makes a huge difference in presentation.

Hope that helps

Z

**Reply by mkende** [reply]
Just to clarify, do you mean a way to format the raw html produced by the library or a way to customise the rendering of that html in the browser ?

**Comment by systemz64** [comment]
So its providing some means to customise the output. For example, if I say all <H1> use style xxxx – that is effectively an over-ride of the default styles that you have.

The alternate is to provide hardcoded “themes” that use different styles – and we are allowed to apply a theme to a convertor at run time?

Just some thoughts
Z

**Comment by mkende** [comment]
There is no style at all in the output of the library, just raw HTML tags. And the output is not a complete document, as it has no header and no body tag. So, what you can do is just add to the returne HTML a `<style>` element or a link to a stylesheet, so that you can provide whatever CSS you want.

Are you using the Perl library or the `pmarkdown` executable?

**Comment by systemz64** [comment]
Ah right. We are using the library
Just pushed to UAT for testing today!

Thank you

Z

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
